crates/services/gas_price_service/src/v1/service.rs

```diff
@@ -136,63 +139,79 @@ where
     }
 
     fn validate_block_gas_capacity(
-        &self,
         block_gas_capacity: u64,
     ) -> anyhow::Result<NonZeroU64> {
         NonZeroU64::new(block_gas_capacity)
             .ok_or_else(|| anyhow!("Block gas capacity must be non-zero"))
     }
 
-    async fn set_metadata(&mut self) -> anyhow::Result<()> {
-        let metadata: UpdaterMetadata = self.algorithm_updater.clone().into();
-        self.metadata_storage
-            .set_metadata(&metadata)
-            .map_err(|err| anyhow!(err))
-    }
-
-    async fn handle_normal_block(
-        &mut self,
+    async fn handle_normal_block<'a>(
```

Comment:
What does "normal" mean in this context? Maybe "non_genesis" would be more clear?

Reply:
As opposed to genesis block. I know it's not a great name, but it's true that normally it will be this block. Haha. I'd be open to other names. "non-genesis" seems strange.

Reply:
Maybe just "handle_block()"?